cmd/kube-apiserver/app/testing/testserver.go

```diff
@@ -222,7 +222,7 @@ func StartTestServer(t Logger, instanceOptions *TestServerInstanceOptions, custo
 		return result, fmt.Errorf("failed to create server chain: %v", err)
 	}
 
-	errCh := make(chan error)
+	errCh := make(chan error, 1)
```

Comment:
I don't think there is a risk of deadlock - putting into errCh is happening in a separate goroutine so everything seems fine to me.

Reply:
The deadlock is only for child goroutine (line 226) and it only happened if line 247 got selected. This fix does not change any test behavior but simply allow proper exit of the child goroutine if line 247 got selected. Similar to #102635 and #83925.